src/arkode/arkode_lsrkstep.c

```diff
@@ -1747,11 +1740,15 @@ int lsrkStep_TakeStepSSP104(ARKodeMem ark_mem, sunrealtype* dsmPtr, int* nflagPt
                      ark_mem->nst, ark_mem->h, ark_mem->tcur);
 #endif
 
-  /* The method is not FSAL, and fn ​is used as storage in each step. Therefore, fn ​is computed at the beginning regardless. */
-  retval = step_mem->fe(ark_mem->tn, ark_mem->yn, ark_mem->fn,
-                        ark_mem->user_data);
-  step_mem->nfe++;
-  if (retval != ARK_SUCCESS) { return (ARK_RHSFUNC_FAIL); }
+  /* The method is not FSAL. Therefore, fn ​is computed at the beginning 
+     of the step unless ARKODE updated fn. */
+  if (!ark_mem->fn_is_current)
+  {
+    retval = step_mem->fe(ark_mem->tn, ark_mem->yn, ark_mem->fn,
+                          ark_mem->user_data);
```

Comment:
Unlike `SSPs2` this function overwrites `fn` during the step so if the temporal error test fails (or the step fails for some other reason) this will use the wrong `fn` value on the next step attempt.